artifacts/openapi.yaml

```diff
@@ -598,206 +610,1872 @@ components:
           minimum: 1
           maximum: 32
           default: 24
-        bgpv4:
-          $ref: '#/components/schemas/Device.Bgpv4'
         name:
           description: |-
             Globally unique name of an object. It also serves as the primary key for arrays of objects.
           type: string
           pattern: ^[\sa-zA-Z0-9-_()><\[\]]+$
-    Device.Bgpv4:
-      x-status: under-review
+    Interface.LoopbackIpv4:
       description: |-
-        Container for emulated BGPv4 peers and routes.
-        Container for basic emulated BGP peer settings.
+        An IPv4 loopback interface
       type: object
       required:
-      - local_address
-      - dut_address
-      - as_type
-      - as_number
-      - name
+      - interface_name
       properties:
-        local_address:
-          description: |-
-            Local IPv4 address of the emulated router
-          type: string
-          format: ipv4
-        dut_address:
-          description: |-
-            IPv4 address of the BGP peer for the session
+        interface_name:
+          description: "The unique name of ethernet object behind which this loopback\
+            \ \n interface will be created.\n\nx-constraint:\n- /components/schemas/Interface.Ethernet/properties/name\n"
           type: string
-          format: ipv4
-        router_id:
+          x-constraint:
+          - /components/schemas/Interface.Ethernet/properties/name
+        address:
           description: |-
-            The BGP router ID is a unique identifier used by routing protocols. It is a 32-bit value that is often represented by an IPv4 address.
+            The Loopback IPv4 address
           type: string
           format: ipv4
           default: 0.0.0.0
```

Comment:
should address be a required property above?  if 0.0.0.0 is not a good default that will work 99% of time, we should make this. a required field.

Reply:
Made required field in PR#154